api/controllers/standard/standard.controller.js

```diff
@@ -1,31 +1,160 @@
+              message: "Invalid / No auth key found",
+            });
+          }
+        }
+      }
+    })
+    .catch((err) => console.log(err));
 };
 
 var get = (req, res, next) => {
-    Standard.find({}, function (err, standards) {
-        if (err)
-            return next(err);
-        res.json(standards)
-    });
+  Standard.find({}, function (err, standards) {
+    if (err) return next(err);
+    res.json(standards);
+  });
+};
+
+const getAllByShortName = async (req, res) => {
+  const shortName = req.body.shortName;
+  await Standard.find({ "info.shortName": shortName })
+    .then((result) => {
+      if (result.length > 0) {
+        result = result[0];
+        res.json(result).status(200);
+      } else {
+        res.json({ status: 404, message: `No data found for ${shortName}` });
+      }
+    })
+    .catch((err) => console.log(err));
+};
+
+const getAllStandardByUUID = async (req, res) => {
+  const uuid = req.body.uuid;
+  await Standard.find({ "uuid": uuid })
+    .then((result) => {
+      if (result.length > 0) {
+        result = result[0];
+        res.json(result).status(200);
+      } else {
+        res.json({ status: 404, message: `No data found for ${uuid}` });
+      }
+    })
+    .catch((err) => console.log(err));
+};
+
+const getLatestStandardByUUID = async (req, res) => {
+    const uuid = req.body.uuid;
+    await Standard.find({ "uuid": uuid })
+      .then((result) => {
+        if (result.length > 0) {
+          result = result[0];
+          result["versions"] = [
+            result["versions"][result["versions"].length - 1],
+          ];
+          res.json(result).status(200);
+        } else {
+          res.json({ status: 404, message: `No data found for ${uuid}` });
+        }
+      })
+      .catch((err) => console.log(err));
+  };
+
+const getLatestByShortName = async (req, res) => {
+  const shortName = req.body.shortName;
+  await Standard.find({ "info.shortName": shortName })
+    .then((result) => {
+      if (result.length > 0) {
+        result = result[0];
+        result["versions"] = [
+          result["versions"][result["versions"].length - 1],
+        ];
+        res.json(result).status(200);
+      } else {
+        res.json({ status: 404, message: `No data found for ${shortName}` });
+      }
+    })
+    .catch((err) => console.log(err));
+};
+
+const updateStandard = async (req, res) => {
+  const shortName = req.body.shortName;
+  // Assuming input versions is an array
+  const versions = req.body.versions[0];
+  // what to update, options -> major, minor, or patch
+  const increment_standard = req.body.increment_standard;
+  await Standard.find({ "info.shortName": shortName })
+    .then(async (result) => {
+      if (result.length > 0) {
+        result = result[0];
+        result["versions"] = [
+          result["versions"][result["versions"].length - 1],
+        ];
+        version = result["versions"][0].version;
+        version[`${increment_standard}`]++;
```

Comment:
We will be following the (semantic versioning standard](https://semver.org) which has 3 parts `MAJOR.MINOR.PATH`. I think the user should pass the kind of update they are doing to the standard (`MAJOR, MINOR or PATCH`) and based on that we should increment the fields.

So, now your `version` object in `standard.model` would look something like this (which is done in this patch):

```json
version: {
  "major": 1,
  "minor": 2,
  "patch": 3
}
```

Based on the `kindOfUpdate` you can increment the fields.

```py
if kindOfUpdate == 'MAJOR':
  new_version = {
    "major": old_version.major + 1,
    "minor": 0,
    "patch": 0
  }
elif kindOfUpdate == 'MINOR':
  new_version = {
    "major": old_version.major,
    "minor": old_version.minor + 1,
    "patch": 0
  }
elif kindOfUpdate == 'PATCH':
  new_version = {
    "major": old_version.major,
    "minor": old_version.minor,
    "patch": old_version.patch + 1
  }
```